hazelcast/src/hazelcast/client/xml_impl.cpp

```diff
@@ -0,0 +1,1440 @@
+const std::vector<std::string> declarative_config_util::XML_ACCEPTED_SUFFIXES = {
+    "xml"
+};
+const std::string declarative_config_util::XML_ACCEPTED_SUFFIXES_STRING = "[xml]";
+const std::vector<std::string> declarative_config_util::ALL_ACCEPTED_SUFFIXES = {
+    "xml",
+    "yaml",
+    "yml"
+};
+const std::string declarative_config_util::ALL_ACCEPTED_SUFFIXES_STRING =
+  "[xml, yaml, yml]";
+
+void
+declarative_config_util::validate_suffix_in_system_property(
+  const std::string& property_key)
+{
+    std::string config_system_property =
+      ""; // System.getProperty(propertyKey); TODO
+    if (config_system_property.empty()) {
+        return;
+    }
+    if (!is_accepted_suffix_configured(config_system_property,
+                                       ALL_ACCEPTED_SUFFIXES)) {
+        throw_unaccepted_suffix_in_system_property(
+          property_key, config_system_property, ALL_ACCEPTED_SUFFIXES);
+    }
+}
+void
+declarative_config_util::throw_unaccepted_suffix_in_system_property(
+  const std::string& property_key,
+  const std::string& config_resource,
+  const std::vector<std::string>& accepted_suffixes)
+{
+
+    std::string message =
+      "The suffix of the resource \'" + config_resource +
+      "\' referenced in \'" + property_key +
+      "\' is not in the list of accepted " + "suffixes: \'[" +
+      boost::algorithm::join(accepted_suffixes, ", ") + "]\'";
+    throw hazelcast::client::exception::hazelcast_(message);
+}
+bool
+declarative_config_util::is_accepted_suffix_configured(
+  const std::string& config_file,
+  std::vector<std::string> accepted_suffixes)
+{
+    std::string config_file_lower;
+    if (config_file.empty()) {
+        config_file_lower = config_file;
+    } else {
+        config_file_lower = config_file;
+        boost::algorithm::to_lower(config_file_lower);
+    }
+    int last_dot_index = (int)config_file_lower.find_last_of('.');
+    if (last_dot_index == -1) {
+        return false;
+    }
+    std::string config_file_suffix =
+      config_file_lower.substr(last_dot_index + 1);
+    return std::find(accepted_suffixes.begin(),
+                     accepted_suffixes.end(),
+                     config_file_suffix) != accepted_suffixes.end();
+}
+
+std::ifstream*
+abstract_config_locator::get_in()
+{
+    return in;
+}
+
+bool
+abstract_config_locator::load_from_working_directory(
+  const std::string& config_file_path)
+{
+    try {
+        std::FILE* file;
+        file = fopen(config_file_path.c_str(), "r");
+        if (file == nullptr) {
+            std::cout << "FINEST: "
+                      << "Could not find " + config_file_path +
+                           " in the working directory."
+                      << std::endl;
+            return false;
+        }
+        std::cout << "INFO: Loading " + config_file_path +
+                       " from the working directory."
+                  << std::endl;
+        configuration_file = file;
+        in = new std::ifstream();
```

Comment:
do not use raw pointer, it is a memory leak. use stack allocated variables instead.